scripts/schemas/sip-schema.yml

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+type: map
```

Comment:
Shouldn't `vendor` field be included in the SiP schema as well?

Reply:
Vendor can be added, it is not strictly neccesary outside of filtering so kept it out for now :)

Reply:
From our perspective this would be an important data point - we treat yaml files not only as configuration for filtering but as a source of information about the hardware. 
The same data can be used for documentation generation, so we are strongly for adding this field to the schema as well.

Reply:
An earlier version of this PR used the sip/<vendor>/ folder name to determine the vendor, I can add it back :) No need to copy the same information to the sip/<vendor>/sip.yml file I believe :)

Reply:
@bjarki-andreasen I wouldn't necessarily say it's copying the same information. We have the exact same case with boards - even though they already are located in `boards/<vendor>`, we still provide the vendor field in `board.yml`. The same holds true for SoCs.
I’d say we should include the vendor field in `sip.yml` because:
* it keeps the format consistent
* we decouple the vendor information from filesystem paths
* it makes data exploration much easier - other than 3rd party tools, it could be for example used by custom Zephyr Sphinx directives for documentation generation and/or other purposes

Reply:
hmm

> it keeps the format consistent

sure :)

> we decouple the vendor information from filesystem paths

The folder name is not arbitrary, it has to be the vendor name, so it is copying information

> it makes data exploration much easier - other than 3rd party tools, it could be for example used by custom Zephyr Sphinx directives for documentation generation and/or other purposes

Tools should not, and do not AFAIK read yaml files directly, tools use CMAKE formatted output from the scripts we use to parse the yaml files, like list_hardware.py and list_boards.py so it makes no difference here I believe